### DISCUSSION TITLE
Can I register custom nodes from within plugins?

### DISCUSSION THREAD

**Discussion by diegohaz** [facebook/lexical#2053] [Q&A] [answered]
I know that I can register my node through the initial config:

```jsx
<LexicalComposer initialConfig={{ nodes: [MyNode] }}>
```

But I'd like to know if it's possible to do that after the first render and inside a plugin.

**Comment by zurfyx** [comment, marked as answer]
All nodes have to registered on composer creation as conditionals nodes can cause the editor to crash or misbehave when trying to restore content on it. In fact, this used to be the previous API  for node registration (`editor.registerNode`).

For example, you copy-paste NodeA created by plugin A. Plugin A is unmounted, NodeA is unregistered and you paste your clipboard contents, editor will crash because NodeA no longer exists.

_Side note: the above is not possible between two Lexical editors living on different surfaces because of the `namespace`, a unique identifier that tells Lexical to fall back to HTML or plain text._

For this reason, you want to register all nodes at once even if you're not using them or might only use later. FWIW, nodes tend to be lightweight while the plugins, which can be deferred, can often be loaded at a later point in time.

**Reply by diegohaz** [reply]
Nice! Thank you for the answer!
